### PR TITLE
Implemented keysHandler

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- implemented keysHandler
  
## Proposed changes
- implemented keysHandler

## Testing procedure
- N/A

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: integrationTests/factory/bootstrapComponents/bootstrapComponents_test.go [hunk 1]
```diff
@@ -33,10 +33,10 @@ func TestBootstrapComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/bootstrapComponents/bootstrapComponents_test.go:38]: should also update the order of `.Close()`, L50 should be switched with L48

File: integrationTests/factory/consensusComponents/consensusComponents_test.go [hunk 2]
```diff
@@ -36,10 +36,10 @@ func TestConsensusComponents_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/consensusComponents/consensusComponents_test.go:41]: same here, managedNetworkComponents should be closed after managedCryptoComponents

File: integrationTests/factory/dataComponents/dataComponents_test.go [hunk 3]
```diff
@@ -29,10 +29,10 @@ func TestDataComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/dataComponents/dataComponents_test.go:34]: managedNetworkComponents should be closed after managedCryptoComponents

File: integrationTests/factory/heartbeatComponents/heartbeatComponents_test.go [hunk 4]
```diff
@@ -36,10 +36,10 @@ func TestHeartbeatComponents_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/heartbeatComponents/heartbeatComponents_test.go:41]: managedNetworkComponents should be closed after managedCryptoComponents

File: integrationTests/factory/networkComponents/networkComponents_test.go [hunk 5]
```diff
@@ -34,11 +34,10 @@ func TestNetworkComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
-	require.NotNil(t, managedNetworkComponents)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/networkComponents/networkComponents_test.go:39]: managedNetworkComponents should be closed after managedCryptoComponents

File: integrationTests/factory/processComponents/processComponents_test.go [hunk 6]
```diff
@@ -37,10 +37,10 @@ func TestProcessComponents_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/processComponents/processComponents_test.go:42]: managedNetworkComponents should be closed after managedCryptoComponents

File: integrationTests/factory/stateComponents/stateComponents_test.go [hunk 7]
```diff
@@ -33,10 +33,10 @@ func TestStateComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/stateComponents/stateComponents_test.go:38]: managedNetworkComponents should be closed after managedCryptoComponents

File: integrationTests/factory/statusComponents/statusComponents_test.go [hunk 8]
```diff
@@ -37,10 +37,10 @@ func TestStatusComponents_Create_Close_ShouldWork(t *testing.T) {
 	require.Nil(t, err)
 	managedStatusCoreComponents, err := nr.CreateManagedStatusCoreComponents(managedCoreComponents)
 	require.Nil(t, err)
-	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents)
-	require.Nil(t, err)
 	managedNetworkComponents, err := nr.CreateManagedNetworkComponents(managedCoreComponents, managedStatusCoreComponents)
 	require.Nil(t, err)
+	managedCryptoComponents, err := nr.CreateManagedCryptoComponents(managedCoreComponents, managedNetworkComponents)
```
Comment (sstanculeanu) [inline integrationTests/factory/statusComponents/statusComponents_test.go:42]: managedNetworkComponents should be closed after managedCryptoComponents

**Review by SebastianMarian — DISMISSED** [review]

File: integrationTests/factory/consensusComponents/consensusComponents_test.go (hunk 2 above)
Comment (iulianpascalau) [reply integrationTests/factory/consensusComponents/consensusComponents_test.go:41]: done

File: integrationTests/factory/dataComponents/dataComponents_test.go (hunk 3 above)
Comment (iulianpascalau) [reply integrationTests/factory/dataComponents/dataComponents_test.go:34]: done

File: integrationTests/factory/heartbeatComponents/heartbeatComponents_test.go (hunk 4 above)
Comment (iulianpascalau) [reply integrationTests/factory/heartbeatComponents/heartbeatComponents_test.go:41]: done

File: integrationTests/factory/processComponents/processComponents_test.go (hunk 6 above)
Comment (iulianpascalau) [reply integrationTests/factory/processComponents/processComponents_test.go:42]: done

File: integrationTests/factory/stateComponents/stateComponents_test.go (hunk 7 above)
Comment (iulianpascalau) [reply integrationTests/factory/stateComponents/stateComponents_test.go:38]: done

File: integrationTests/factory/statusComponents/statusComponents_test.go (hunk 8 above)
Comment (iulianpascalau) [reply integrationTests/factory/statusComponents/statusComponents_test.go:42]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**